deploy/os/controller.yaml

```diff
@@ -0,0 +1,72 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: che-workspace-controller
+  namespace: che-workspace-controller
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: che-workspace-controller
+  template:
+    metadata:
+      labels:
+        app: che-workspace-controller
+    spec:
+      serviceAccountName: che-workspace-controller
+      containers:
+        - name: che-workspace-controller
+          image: quay.io/che-incubator/che-workspace-controller:nightly
+          imagePullPolicy: Always
+          env:
+            - name: WATCH_NAMESPACE
+              value: ""
+            - name: POD_NAME
+              valueFrom:
+                fieldRef:
+                  fieldPath: metadata.name
+            - name: OPERATOR_NAME
+              value: "che-workspace-operator"
+          ports:
+            - name: webhook-server
+              containerPort: 8443
+          volumeMounts:
+            - name: webhook-tls-certs
+              mountPath: /tmp/k8s-webhook-server/serving-certs
+              readOnly: true
+      volumes:
+        - name: webhook-tls-certs
+          projected:
+            sources:
+              - configMap:
+                  name: che-workspace-controller-secure-service
```

Comment:
Will deployment be triggered when OpenShift rotates sources?
if no - maybe it makes sense to watch this configmap and secret from WorkspaceController
Or maybe there is another mechanism on how to restart the controller in such case.

Reply:
We have a mechanism in place for watching the main controller configmap (e.g. [here](https://github.com/amisevsk/che-workspace-crd-operator/blob/dde776de73562554987d33854d51bdf60f13226a/pkg/config/config.go#L141)) but I don't know if it's suitable if we need to *restart* the controller pod. I don't know that controllers typically roll out their own deployments, especially since this could look like an issue in the controller itself.

Reply:
> Will deployment be triggered when OpenShift rotates sources?

The documentation as far as I can tell doesn't say what would happen when the sources are rotated. All it says is:
> The certificate and key are automatically replaced when they get close to expiration

I'll probably have to dig a little bit deeper into that code to see what would happen

Reply:
![Screenshot_20200402_123226](https://user-images.githubusercontent.com/5887312/78233203-0eea0180-74de-11ea-9ae3-a12657b51774.png)

It makes me thinking that instead of mounting files we could:
- watch secret - at the starttime - initialize files on the filesystem with the current values.
  - once secret is modified - update the files content. Probably cert watcher will get updated values and restart webbhook server - it's needed to be checked.
- watch configmap - update WebhooksCfgs when certs are changed in configmap

But since by default certificates has 1-year expiration - we could postpone this task.